src/core/jwt/jwtTokenStore.js

```diff
@@ -55,6 +59,11 @@ const jwtTokenStoreFactory = function jwtTokenStoreFactory({
          * @returns {Promise<Boolean>} token successfully set
          */
         setAccessToken(token) {
+            if (usePerTokenTTL) {
+                const tokenPayload = parseJwtPayload(token);
+                const buffer = 10000;
+                currentAccessTokenTTL = (getJwtTTL(tokenPayload) || defaultAccessTokenTTL) - buffer;
```

Comment:
suggestion: would it be safer to ensure the value cannot be negative (in case of misconfiguration of the  defaultAccessTokenTTL to `0` for example)
```suggestion
                currentAccessTokenTTL = Math.max((getJwtTTL(tokenPayload) || defaultAccessTokenTTL) - buffer, 0);
```

Reply:
I don't think the behaviour will be any different whether we are checking `accessTokenStoredAt < Date.now()` or `accessTokenStoredAt - 10000 < Date.now()`. Both will fail in the event of a misconfigured defaultAccessTokenTTL.

But I can agree to wrap it for the sake of more readable code.